cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java

```diff
@@ -44,51 +46,79 @@ public class CruiseControlApiImpl implements CruiseControlApi {
      * Default timeout for the HTTP client (-1 means use the clients default)
      */
     public static final int HTTP_DEFAULT_IDLE_TIMEOUT_SECONDS = -1;
-    private static final boolean HTTP_CLIENT_ACTIVITY_LOGGING = false;
     private static final String STATUS_KEY = "Status";
-
-    private final Vertx vertx;
     private final long idleTimeout;
     private final boolean apiSslEnabled;
     private final HTTPHeader authHttpHeader;
-    private final PemTrustOptions pto;
-
+    private final PemTrustSet pemTrustSet;
+    private final HttpClient httpClient;
     /**
      * Constructor
      *
-     * @param vertx             Vert.x instance
      * @param idleTimeout       Idle timeout
      * @param ccSecret          Cruise Control Secret
      * @param ccApiSecret       Cruise Control API Secret
      * @param apiAuthEnabled    Flag indicating if authentication is enabled
      * @param apiSslEnabled     Flag indicating if TLS is enabled
      */
-    public CruiseControlApiImpl(Vertx vertx, int idleTimeout, Secret ccSecret, Secret ccApiSecret, Boolean apiAuthEnabled, boolean apiSslEnabled) {
-        this.vertx = vertx;
+    public CruiseControlApiImpl(int idleTimeout, Secret ccSecret, Secret ccApiSecret, Boolean apiAuthEnabled, boolean apiSslEnabled) {
         this.idleTimeout = idleTimeout;
         this.apiSslEnabled = apiSslEnabled;
         this.authHttpHeader = getAuthHttpHeader(apiAuthEnabled, ccApiSecret);
-        this.pto = new PemTrustOptions().addCertValue(Buffer.buffer(Util.decodeBase64FieldFromSecret(ccSecret, "cruise-control.crt")));
+        this.pemTrustSet = new PemTrustSet(ccSecret);
+        this.httpClient = buildHttpClient();
     }
 
     @Override
-    public Future<CruiseControlResponse> getCruiseControlState(Reconciliation reconciliation, String host, int port, boolean verbose) {
-        return getCruiseControlState(reconciliation, host, port, verbose, null);
-    }
+    public CompletableFuture<CruiseControlResponse> getCruiseControlState(Reconciliation reconciliation, String host, int port, boolean verbose) {
+        String path = new PathBuilder(CruiseControlEndpoints.STATE)
+                .withParameter(CruiseControlParameters.VERBOSE, String.valueOf(verbose))
+                .withParameter(CruiseControlParameters.JSON, "true")
+                .build();
 
-    private HttpClientOptions getHttpClientOptions() {
-        if (apiSslEnabled) {
-            return new HttpClientOptions()
-                .setLogActivity(HTTP_CLIENT_ACTIVITY_LOGGING)
-                .setSsl(true)
-                .setVerifyHost(true)
-                .setTrustOptions(
-                    new PemTrustOptions(pto)
-                );
-        } else {
-            return new HttpClientOptions()
-                    .setLogActivity(HTTP_CLIENT_ACTIVITY_LOGGING);
+        HttpRequest.Builder builder;
+        builder = HttpRequest.newBuilder()
```

Comment:
just a nit ... why two lines (one for declaration and the other for instantiation)?
The same applies in all the other places where you use a builder.